lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableRemoteEvent.cs

```diff
@@ -72,7 +78,7 @@ public WritableRemoteEvent()
         /// <param name="reader">The reader from which to read </param>
         public void Read(IDataReader reader)
         {
-            Value = Activator.CreateInstance<T>();
+            Value = (T) _injector.ForkInjector().GetInstance(typeof(T));
```

Comment:
Are you expecting different instance every time this method is called even T is the same?